File: cdb/rhino/read/cdb_read.js
#! /usr/bin/java org.mozilla.javascript.tools.shell.Main
// ----------------------------------------------------------------------- 
/*
	cdb_read.js

					Jun/17/2015

*/
// ----------------------------------------------------------------------- 
importPackage (com.strangegizmo.cdb)

load ("/var/www/data_base/common/rhino_common/text_manipulate.js")
// ----------------------------------------------------------------------- 
print ("*** 開始 ***")

file_cdb=arguments[0]

print (file_cdb)

ee = Cdb.elements (file_cdb)

var dict_aa = new Object ()

while (ee.hasMoreElements())
	{
	element = ee.nextElement()
	key_byte = element.getKey()
	key = new java.lang.String (key_byte)
	data_byte = element.getData()
	str_json = new java.lang.String (data_byte)

	unit_aa = JSON.parse (str_json)
	dict_aa[key] = unit_aa
	}

dict_display_proc (dict_aa)

print ("*** 終了 ***")
// ----------------------------------------------------------------------- 

File: cdb/java/read/cdb_read.java
// ----------------------------------------------------------------
/*
	cdb_read.java

					May/29/2013

*/
// ----------------------------------------------------------------
import java.io.IOException;
import java.util.HashMap;

// ----------------------------------------------------------------
public class cdb_read
{
// ----------------------------------------------------------------
public static void main(String[] args) throws IOException
{
	System.out.println ("*** 開始 ***");

	String file_cdb = args[0];

	HashMap <String, HashMap<String,String>> dict_aa
		= cdb_manipulate.cdb_to_dict_proc (file_cdb);

	text_manipulate.dict_display_proc (dict_aa);
		
	System.out.println ("*** 終了 ***");
}

// ----------------------------------------------------------------
}
// ----------------------------------------------------------------
